### PR TITLE
#143412445 Convert grunt to gulp 

### PR DESCRIPTION
**Pull request by muchai-mercy**
#### What does this PR do?
Converts Grunt to Gulp

 #### Description of Task to be completed?
Run mocha to check tests
Run eslint to check similar code conventions
Run nodemon to watch CFH is running
Run sass to check writing styles
Run browserSync to sync file changes and app on browser

 #### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2003755/stories/143412445

#### Screenshot
screen shot 2017-04-19 at 4 30 34 pm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ c4h_keys.txt
 .idea/vcs.xml
 Project_Default.xml
 .env
+gruntfile.js
```
Comment (gngchrs) [inline .gitignore:9]: gruntfile is supposed to be deleted, not ignored

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ext: 'js html',
+    env: { NODE_ENV: 'development' } });
+});
+// sass task
+gulp.task('sass', () => gulp.src('./sass/**/*.scss')
+        .pipe(sass().on('error', sass.logError))
+        .pipe(gulp.dest('.public/css')));
+gulp.task('sass:watch', () => {
+  gulp.watch('./sass/**/*.scss', ['sass']);
+});
```
Comment (gngchrs) [inline gulpfile.js:32]: minor: have some space between the gulp tasks

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+gulp.task('watch', () => {
+  gulp.watch(['app/**/*.js', 'public/js/**/*.js'], ['lint'])
+    .on('change', browserSync.reload);
+});
+gulp.task('mocha', () => {
```
Comment (gngchrs) [inline gulpfile.js:37]: ditto: see above

**Comment by gngchrs** [comment]
I also notice  2 things 

- `package.json` still has `grunt commands`

- You don't have a bower task, like we had discussed.

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// Include gulp
+const gulp = require('gulp');
+const gutil = require('gulp-util');
+// Include gulp plugins
+const eslint = require('gulp-eslint');
+const nodemon = require('gulp-nodemon');
+const sass = require('gulp-sass');
+const browserSync = require('browser-sync');
+const mocha = require('gulp-mocha');
```
Comment (gngchrs) [inline gulpfile.js:9]: never used. ESLint should capture this. @andela-wmumbi hound should capture this.

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        .pipe(gulp.dest('.public/css')));
+gulp.task('sass:watch', () => {
+  gulp.watch('./sass/**/*.scss', ['sass']);
+});
+gulp.task('watch', () => {
+  gulp.watch(['app/**/*.js', 'public/js/**/*.js'], ['lint'])
+    .on('change', browserSync.reload);
+});
+gulp.task('mocha', () => {
+  gulp.src('test.js', { read: false });
```
Comment (gngchrs) [inline gulpfile.js:38]: I'm not sure this task does anything. Does it?

File: package.json [hunk 6]
```diff
@@ -13,9 +13,8 @@
     "npm": "4.2.0"
   },
   "scripts": {
-    "start": "node node_modules/grunt-cli/bin/grunt",
-    "test": "node node_modules/grunt-cli/bin/grunt test",
-    "postinstall": "node node_modules/grunt-cli/bin/grunt install"
+    "start": "gulp",
+    "test": "gulp mocha"
```
Comment (gngchrs) [inline package.json:17]: We need a postinstall here to run bower. 
```
"postinstall": "gulp bower"
```

Comment (gngchrs) [inline package.json:17]: `gulp mocha` doesn't seem to do anything. Just remove it. I saw @Ewanjiru has an istanbul version for npm test.

File: .gitignore (hunk 1 above)
Comment (gngchrs) [reply .gitignore:9]: remove this and delete the grunt file.

**Comment by gngchrs** [comment]
Oh, we also need a `.bowerrc` file with the following contents
```
{
    "directory":"./public/lib"
}
```

**Comment by gngchrs** [comment]
I also notice after running bower and installing things, this is the error that occurs.

screen shot 2017-04-19 at 9 36 10 pm

These files are not on the correct path. For instance, the app calls `/lib/bootstrap/js/bootstrap.js` while the actual file is in `/lib/bootstrap/dist/js/bootstrap.js`.

You can find where to make the correct updates in `foot.jade` and `head.jade`

File: gulpfile.js (hunk 5 above)
Comment (muchai-mercy) [reply gulpfile.js:38]: Yeah. Realised it was to run tests on all .js files in test directory

**Pull request closed without merging**